canonicalwebteam/blog_extension/logic.py

```diff
@@ -0,0 +1,156 @@
+import re
+import html
+
+from datetime import datetime
+
+
+def strip_excerpt(raw_html):
+    """Remove tags from a html string
+
+    :param raw_html: The HTML to strip
+
+    :returns: The stripped string
+    """
+    cleanr = re.compile("<.*?>")
+    cleantext = re.sub(cleanr, "", raw_html)
+    return html.unescape(cleantext).replace("\n", "")
+
+
+def replace_images_with_cloudinary(content):
+    """Prefixes images with cloudinary optimised URLs and adds srcset for
+    image scaling
+
+    :param content: The HTML string to convert
+
+    :returns: Update HTML string with converted images
+    """
+    cloudinary = "https://res.cloudinary.com/"
+
+    urls = [
```

Comment:
Not the most important point (not a blocker), but the list seems more confusing than simply variables. E.g.:

``` python3
url_350 = cloudinary + "canonical/image/fetch/q_auto,f_auto,w_350/\g<url>"
url_650 = cloudinary + "canonical/image/fetch/q_auto,f_auto,w_650/\g<url>"
url_1300 = cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1300/\g<url>"
url_1950 = cloudinary + "canonical/image/fetch/q_auto,f_auto,w_1950/\g<url>"
```

(I may have written this initially, so I'm probably arguing with myself)

Reply:
Or how about:

``` python3
    image_prefix = cloudinary + "canonical/image/fetch/q_auto,f_auto"

    image_match = (
        r'<img(?P<prefix>[^>]*) src="(?P<url>[^"]+)"(?P<suffix>[^>]*)>'
    )
    replacement = (
        "<img\g<prefix>"
        f' decoding="async"'
        f' src="{image_prefix},w_350"'
        f' srcset={image_prefix},w_350 350w, {image_prefix},w_650 650w,'
        f' {image_prefix},w_1300 1300w, {image_prefix},w_1950 1950w"'
        f' sizes="(max-width: 400px) 350w, 650px"'
        "\g<suffix>>"
    )
```

Reply:
Okay issue filed for this: https://github.com/canonical-webteam/blog-flask-extension/issues/3

Resolving this thread.

Reply:
as discussed, this should be an issue for now